Packages/com.unity.inputsystem/InputSystem/Plugins/EnhancedTouch/TouchSimulation.cs

```diff
@@ -280,11 +290,14 @@ protected void OnDisable()
 
         private unsafe void UpdateTouch(int touchIndex, int pointerIndex, TouchPhase phase, InputEventPtr eventPtr = default)
         {
-            var position = m_CurrentPositions[pointerIndex];
+            Vector2 position = m_CurrentPositions[pointerIndex];
+            byte displayIndex = (byte)m_CurrentDisplayIndices[pointerIndex];
```

Comment:
Since this is a narrowing conversion I would suggest adding a debug assert here to just catch an unlikely value larger than byte max value. Otherwise such an unfortnate situation might just be silently truncated.